src/node-file-trace.ts

```diff
@@ -355,6 +355,7 @@ export class Job {
           await this.emitFile(asset, 'asset', path);
       }),
       ...[...deps].map(async dep => {
+        const queryString = dep.split("?")[1]
```

Comment:
Is there a scenario where a file can contain a `?` and dropping it would fail resolution?

I know that CJS needs to keep supporting it but I'm wondering if we need proper URL parsing to match how ESM works.